### ISSUE TITLE
v1.15.1 does not open WikiHow articles

### ISSUE THREAD

**Issue by Popolechien** [kiwix/apple#422]
From a user running on iOS and using wikihow_en_maxi_2021-10.zim

Articles look blanked
![attachment](https://user-images.githubusercontent.com/21675885/144041451-cdc01af9-0e0f-4d84-80b7-c2e5ae09bc4e.png)

Search works, but then every entry opened returns the blank screen.
![attachment-1](https://user-images.githubusercontent.com/21675885/144041517-56ebcd5a-df86-4722-ba41-52f1d8aa75b0.png)

the user tested and this issue is not present on macOS.

**Comment by automactic** [comment]
When getting the main page url, the path returned from libkiwix is `mainPage`. For individual articles in search results, the url path I got contains a lot of percent encoded values. For example: `%D8%B3%D8%B1%D8%B9%D8%AA-%D9%85%DB%8C%D8%A7%D9%86%DA%AF%DB%8C%D9%86-%D8%B1%D8%A7-%D9%85%D8%AD%D8%A7%D8%B3%D8%A8%D9%87-%DA%A9%D9%86%DB%8C%D9%85"	0x0000000280318af0`.

In all of these scenarios, when asking `kiwix::Reader` object for content using `getEntryFromPath`, `kiwix::NoEntry` is raised.

**Comment by automactic** [comment]
@kelson42 Since I cannot get a valid url with content from libkiwix, I am gonna have to ask you to investigate. If you think I should update libkiwix, I could try that

**Comment by kelson42** [comment]
@automactic Wikihow ZIM files are "new" ZIM files, which means built with libzim7. See https://libzim.readthedocs.io/en/latest/6to7.html for more details.

To a large extend, our Wikihow ZIM files are readable witk "older" readers and I don't have seen the symtom we have here. That said, this is highly probable that this is a kind of incompatibility with this new kind of ZIM file.

I would recommend to try to build against the latest libkiwix (still in dev) and adapt the code to new API. Pretty sure this is the best approach tonsolve this problem.

**Comment by automactic** [comment]
@kelson42 I rebuilt libkiwix with kiwix-build, the same issue persists. I suggest transfer this issue to libkiwix

**Comment by kelson42** [comment]
Yes, already done, see https://github.com/kiwix/libkiwix/issues/643

**Comment by kelson42** [comment]
I still believe this is something around kiwix and not libkiwix as this ZIM file works well with kiwix-desktop and kiwix-serve. @mgautierfr Any idea?

**Comment by automactic** [comment]
> I still believe this is something around kiwix and not libkiwix as this ZIM file works well with kiwix-desktop and kiwix-serve.

@kelson42 By "something around kiwix", do you mean the kiwix for iOS app? I am not sure there is anything I can do if the url path I got from libkiwix cannot be loaded.

**Comment by rgaudin** [comment]
Looks like an url-encoding issue to me. If you look closely at those links, you'll realize that in the HTML source code (*right-click -> Copy Link* to retrieve), links are urlencoded as in your example above. But when you over the link, you'll notice your browser converts it and displays the decoded string.

[this is your example link](http://library.kiwix.org/wikihow_fa_maxi/C/%D8%B3%D8%B1%D8%B9%D8%AA-%D9%85%DB%8C%D8%A7%D9%86%DA%AF%DB%8C%D9%86-%D8%B1%D8%A7-%D9%85%D8%AD%D8%A7%D8%B3%D8%A8%D9%87-%DA%A9%D9%86%DB%8C%D9%85)

Obviously, ZIM paths are stored decoded so I bet that if you tried to access the ZIM path using the decoded string it would work. Could you try with this example ?

```py
>>> import urllib.parse
>>> urllib.parse.unquote("%D8%B3%D8%B1%D8%B9%D8%AA-%D9%85%DB%8C%D8%A7%D9%86%DA%AF%DB%8C%D9%86-%D8%B1%D8%A7-%D9%85%D8%AD%D8%A7%D8%B3%D8%A8%D9%87-%DA%A9%D9%86%DB%8C%D9%85")
'سرعت-میانگین-را-محاسبه-کنیم'
```

**Comment by mgautierfr** [comment]
If wikihow_fa (and in wikihow_en) the path of the item is utf8 encoded.

What is stored in the zim is byte array of `[0xD8, 0xB3, 0xD8, 0xB1, ...]` and this is what is returned by libzim and libkiwix.
If you need unicode, you must utf8 decode the bytes to have the unicode path.

---

I don't know what is the problem for wikihow_en, but I've download it and it works well with kiwix-serve.

**Comment by kelson42** [comment]
Ok, so I assume the `libkiwix` build is the one of the git master HEAD.

You have to call `getEntryFromPath("سرعت-میانگین-را-محاسبه-کنیم")` (that means the URL encoded value, but decoded) that should not return `kiwix::NoEntry`. If it does not, but would be really surprising, because it works in kiwix-serve. Then we could have a look more in detail.

**Comment by automactic** [comment]
Version 1.15.3 will be reverting to the last known stable version of libkiwix and libzim. It is currently waiting for review. So I will consider this matter closed if no one here raises objection in the next few days.

@kelson42 If you think there are issue that needs to be fixed on libkiwix side, please transfer this ticket or open new ones

**Comment by kelson42** [comment]
Would be better to answer our questions to understand what goes on... you will anyway have to move forward with libkiwix10 or later at some point.

**Comment by automactic** [comment]
What specifically do you need me to answer?

**Comment by kelson42** [comment]
@mgautierfr Why calling getEntryFromPath("سرعت-میانگین-را-محاسبه-کنیم") returns an exception with libkiwix10 and wikihow_fa_maxi_2021-10.zim on iOS whereas it works on linux?
